app/tasks/vector_source_assets.py

```diff
@@ -60,48 +59,88 @@ async def vector_source_asset(
         callback=callback,
     )
 
-    # AWS Batch jobs can't have more than 20 parents. In case of excessive
-    # numbers of layers, create multiple "queues" of dependent jobs, with
-    # the next phase being dependent on the last job of each queue.
-    num_queues = min(16, len(layers))
-    job_queues: RingOfLists = RingOfLists(num_queues)
     load_vector_data_jobs: List[GdalPythonImportJob] = list()
-    for i, layer in enumerate(layers):
-        queue = next(job_queues)
-        if not queue:
-            parents: List[str] = [create_vector_schema_job.job_name]
-        else:
-            parents = [queue[-1].job_name]
-        job = GdalPythonImportJob(
-            dataset=dataset,
-            job_name=f"load_vector_data_layer_{i}",
-            command=[
-                "load_vector_data.sh",
+    if creation_options.source_driver == VectorDrivers.csv:
+        chunk_size = math.ceil(len(source_uris) / BATCH_DEPENDENCY_LIMIT)
+        uri_chunks = [
+            source_uris[x : x + chunk_size]
+            for x in range(0, len(source_uris), chunk_size)
+        ]
+
+        for i, uri_chunk in enumerate(uri_chunks):
+            command = [
+                "load_vector_csv_data.sh",
                 "-d",
                 dataset,
                 "-v",
                 version,
-                "-s",
-                source_uri,
-                "-l",
-                layer,
-                "-f",
-                local_file,
-                "-X",
-                str(zipped),
-            ],
-            parents=parents,
-            environment=job_env,
-            callback=callback,
-        )
-        queue.append(job)
-        load_vector_data_jobs.append(job)
+            ]
+
+            for uri in uri_chunk:
+                command.append("-s")
+                command.append(uri)
+
+            job = GdalPythonImportJob(
+                dataset=dataset,
+                job_name=f"load_vector_data_layer_{i}",
+                command=command,
+                parents=[create_vector_schema_job.job_name],
+                environment=job_env,
+                callback=callback,
+            )
+            load_vector_data_jobs.append(job)
+
+        load_data_parents = [job.job_name for job in load_vector_data_jobs]
+    else:
+        if len(source_uris) != 1:
+            raise AssertionError(
+                "Non-CSV vector sources require one and only one input file"
+            )
```

Comment:
It would actually be best to check this as early as possible to give users immediate feedback.
You could make this a `validator` in the VectorSourceCreationOptions 


```python
@validator("source_uri")
 def validate_source_uri(cls, v, values, **kwargs):
        if values.get("source_driver") != VectorDrivers.csv:
            assert len(v), "Non-CSV vector sources require one and only one input file"
        return v
```